tools/ci_build/gen_def.py

```diff
@@ -57,6 +57,7 @@ def parse_arguments():
    file.write("#include <onnxruntime_c_api.h>\n")
    for c in args.config:
       file.write("#include <core/providers/%s/%s_provider_factory.h>\n" % (c,c))
+   file.write("#include <adapter/WinMLAdapterFactory.h>\n")
```

Comment:
IMHO, this export should not exist on Linux.

The def file is sufficient for exporting these methods on windows.

It seems like having this be part of generate_sources.c, the export will winml adapter on Linux which seems wrong.

For now, maybe it makes sense to only include it as part of the def file, as I have been told this will not export anything on Linux if it is not additionally referenced by some dll sources. Then we follow up to no build/link winml adapter on non-windows platform builds.

Reply:
Yep I agree! I've added an additional conditional check to exclude WinML in generated_sources.c